src/main/resources/XWiki/SuggestDisplay.xml

```diff
@@ -258,13 +264,52 @@
     img.observe('click', this.removeItem.bindAsEventListener(this));
     li.appendChild(span);
     li.appendChild(hiddenInput);
+    //Add the edit button
+    if(this.inEditItemsMode($(this.list).up(0))){
+        var editImg = new Element("img",{'src': "$xwiki.getSkinFile('icons/silk/page_white_edit.png')"});
+        editImg.observe('click', this.editItem.bindAsEventListener(this));
+        li.appendChild(editImg);
+    }
     li.appendChild(img);
+
     this.list.appendChild(li);
     
     // re-setup the list drag and drop so that the new element can also be moved
     this.setupDragDrop();
   },
 
+  //Check if in items editable mode
+  inEditItemsMode : function(elem){
+    var rep = false;
+    var flag = Element.select(elem, ".inEditItemsMode");
+    if(flag.length &gt; 0){//In items edit mode
+       rep = true;
+    }
+    return rep;
+  },
+
+  editItem : function(event) {
+    var li = event.findElement('li');
+    var input = li.select("input[type='hidden']");
+    var itemDocRef = input[0].value;
+    //Get item class name and property name
+    require(['jquery'],function($){
+       var className = $("input[type='hidden'][id$='_className']",$(li).parent().parent()).eq(0).val();
```

Comment:
Why do you need jQuery there? Is it something you cannot do with prototype?

Actually, since Prototype is deprecated now, the better would be to rewrite everything with jQuery, but I agree it's beyond the scope of this pull request :)